service/src/main/kotlin/fi/espoo/evaka/holidayperiod/HolidayPeriodControllerCitizen.kt

```diff
@@ -43,12 +49,38 @@ class HolidayPeriodControllerCitizen(private val accessControl: AccessControl) {
     @GetMapping("/questionnaire")
     fun getActiveQuestionnaires(
         db: Database,
-        user: AuthenticatedUser,
+        user: AuthenticatedUser.Citizen,
         evakaClock: EvakaClock,
-    ): List<FixedPeriodQuestionnaire> {
+    ): List<FixedPeriodQuestionnaireWithChildren> {
         Audit.HolidayPeriodsList.log()
         accessControl.requirePermissionFor(user, Action.Global.READ_ACTIVE_HOLIDAY_QUESTIONNAIRES)
-        return db.connect { dbc -> dbc.read { listOfNotNull(it.getActiveFixedPeriodQuestionnaire(evakaClock.today())) } }
+        return db.connect { dbc ->
+            dbc.read { tx ->
+                val activeQuestionnaire = tx.getActiveFixedPeriodQuestionnaire(evakaClock.today())
+                if (activeQuestionnaire == null) listOf()
+                else {
```

Comment:
nitpicking: nesting with else is redundant and less readable

Reply:
👍